### ISSUE TITLE
RestAdapter#_stripIDFromURL() fails to behave properly when model.id has url-encodable characters

### ISSUE THREAD

**Issue by aroberts** [emberjs/data#4479]
I found an issue with the `coalesceFindRequests` behavior in a bit of a weird edge case. I have a model that uses phone numbers as ids, so each id starts with `"+"`. I couldn't get the coalescing behavior to fire, so after stepping through with a debugger, I found the issue:

`_stripIDFromURL()` checks for `lastSegment === id`, but not `lastSegment === encodeURIComponent(id)`, so my segments weren't matching the ids.

My fix for now was just to override the method with a copy that encodes the id before checking:

```javascript
  _stripIDFromURL(store, snapshot) {
    var url = this.buildURL(snapshot.modelName, snapshot.id, snapshot);

    var expandedURL = url.split('/');
    //Case when the url is of the format ...something/:id
    var lastSegment = expandedURL[expandedURL.length - 1];
    var id = encodeURIComponent(snapshot.id);
    if (lastSegment === id) {
      expandedURL[expandedURL.length - 1] = "";
    } else if (endsWith(lastSegment, '?id=' + id)) {
      //Case when the url is of the format ...something?id=:id
      expandedURL[expandedURL.length - 1] = lastSegment.substring(0, lastSegment.length - id.length - 1);
    }

    return expandedURL.join('/');
  }
```

**Comment by pangratz** [comment]
Thanks for reporting @aroberts. This should be fixed with #4403. What version are you using? Can you check with `v2.7.0-beta.1` and see if this bug is fixed?

**Comment by aroberts** [comment]
I'm running 2.6.1. I don't have a good way to boot 2.7.0-b1 right now but from the code in the PR it looks like this would fix my issue, and that this issue is a dup of #4190. Thanks!
